pkg/kv/kvserver/kvserverbase/nondeterministic_error.go

```diff
@@ -18,53 +18,53 @@ import (
 	"github.com/cockroachdb/redact"
 )
 
-// NonDeterministicFailureError is an error type that indicates that a state machine
+// NonDeterministicError is an error type that indicates that a state machine
 // transition failed due to an unexpected error. Failure to perform a state
 // transition is a form of non-determinism, so it can't be permitted for any
 // reason during the application phase of state machine replication. The only
 // acceptable recourse is to signal that the replica has become corrupted.
 //
-// All errors returned by replicaDecoder and replicaStateMachine will be instances
-// of this type.
-type NonDeterministicFailureError struct {
+// Many errors returned by `apply.Decoder` and apply.StateMachine implementations
```

Comment:
supernit: use or not use back-ticks consistently (I think the default Go commenting recommends not using them, but lazy to find a link :))

(feel free to ignore or incorporate when merging fixups anyway)